tests/bot_tests/SOA_test.py

```diff
@@ -110,7 +111,55 @@ def test_stance(self):
         print('finish test_stance')
         stop_io_loop()
 
-        # to do: add action-based stance test
+    @gen.coroutine
+    def test_action_stance(self):
+        # score-based
+        game = Game()
+        soa_bot = SmartOrderAccepterBot('FRANCE', game)
+        bot_instances = [RandomProposerBot_AsyncBot('ENGLAND', game), RandomProposerBot_AsyncBot('GERMANY', game), soa_bot]
+        game_play = GamePlay(game, bot_instances, 3, True)
+        game_play.game.set_orders('FRANCE', ['A MAR H', 'A PAR H', 'F BRE - PIC'])
+        game_play.game.set_orders('ENGLAND', ['A LVP - WAL', 'F EDI - NTH', 'F LON - ENG'])
+        game_play.game.set_orders('GERMANY', ['A BER - MUN', 'A MUN - BUR', 'F KIE - HOL'])
+        game_play.game.process()
+        game_play.game.set_orders('FRANCE', ['A MAR - BUR', 'A PAR - BRE', 'F PIC H'])
+        game_play.game.set_orders('ENGLAND', ['A WAL - BEL VIA', 'F ENG C A WAL - BEL', 'F NTH - HEL'])
+        game_play.game.set_orders('GERMANY', ['A BUR - MAR', 'A MUN - RUH', 'F HOL H'])
+        soa_bot_stance = soa_bot.stance.get_stance()[soa_bot.power_name]
+        print(soa_bot_stance)
```

Comment:
I will remove this print out when Tony fix score based stance and I complete the test